src/components/pages/product/ProductPage.ts

```diff
@@ -0,0 +1,163 @@
+import Page from '../../common/Page';
+import './product.css';
+
+class ProductPage extends Page {
+  static TextObject = {
+    MainTitle: 'Settings Page',
+  };
+  private id: string | undefined;
+
+  constructor(id: string) {
+    super(id);
+    this.id = id;
+  }
+
+  async getFetch(dataUrl: string) {
+    return fetch(dataUrl)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        return data;
+      });
+  }
+
+  async createProductContainer() {
+    const id = this.id?.split('/')[1];
+    const dataUrl = 'https://dummyjson.com/products/' + id;
+    const productInfo = await this.getFetch(dataUrl);
+    const productContainer = document.createElement('div');
+    productContainer.classList.add('product-container');
+
+    const productTitle = productInfo.title;
+    const productDescription = productInfo.description;
+    const productCategory = productInfo.category;
+    const productBrand = productInfo.brand;
+    const productDiscount = productInfo.discountPercentage;
+    const productImages = productInfo.images;
+    const productPrice = productInfo.price;
+    const productRating = productInfo.rating;
+    const productStock = productInfo.stock;
```

Comment:
вместо нагромождения констант используем деструкторизацию
```suggestion
   const { title, description, category, brand ..... } = productInfo;
```

Reply:
Исправлено